### DISCUSSION TITLE
Is it possible to write a Ini parser?

### DISCUSSION THREAD

**Discussion by ShuiRuTian** [zesterer/chumsky#646] [Q&A] [answered]
In examples, it usually has a tree-like object. Is it possible to write an Ini parser? The main concern is about the array syntax and section-merge, like:

``` ini

[section1]
foo[]=1

[section2]
bar=2

[section1]
foo[]=3

```

**Comment by ShuiRuTian** [comment]
For now, I use chumksy to write a lexer to parse each line as one token and then collect all info to build the ini.

It works well, still wonder is there any preferred way?

**Comment by zesterer** [comment, marked as answer]
Sure! Parsing this should be possible. Note that things like section merging probably shouldn't be done in the parser anyway (one can imagine a parser that does this, but it's a bit unclear to throw it all in the same place).
